internal/fsutil/create.go

```diff
@@ -121,6 +130,28 @@ func createSymlink(o *CreateOptions) error {
 	return os.Symlink(o.Link, o.Path)
 }
 
+func createHardLink(o *CreateOptions) error {
+	debugf("Creating hard link: %s => %s", o.Path, o.Link)
+	targetInfo, err := os.Lstat(o.Link)
+	if err != nil && os.IsNotExist(err) {
+		return fmt.Errorf("link target does not exist: %s", o.Link)
+	} else if err != nil {
+		return err
+	}
+
+	linkInfo, err := os.Lstat(o.Path)
+	if err == nil || os.IsExist(err) {
+		if os.SameFile(targetInfo, linkInfo) {
+			return nil
+		}
```

Comment:
I did some more tests and `ln` does not respect this, let's remove it. Probably previous comment about naming does not apply as we no longer need those variables.

Reply:
Bookkeeping: As per the offline discussion, this has to exist since there can be different slices in one package that create the same hard link multiple times.